### PR TITLE
LPD-35070

### PR DESCRIPTION
**Pull request by SelenaAungst**
https://liferay.atlassian.net/browse/LPD-35070

## Goal
To provide the theme selection when creating a style book

## Approach
Retrieve the list of available frontend token definitions and their theme ids.

## Limitation
This task does not aim to show the frontend tokens available for the chosen theme. That feature will be implemented at https://liferay.atlassian.net/browse/LPD-35071

Resent from https://github.com/liferay-platform-experience/liferay-portal/pull/791

cc @thiago-buarqque @ethib137 @algobob @matuzalemsteles

### CODE REVIEW

**Comment by SelenaAungst** [comment]
ci:test:sf

**Comment by SelenaAungst** [comment]
ci:test:sf

**Comment by SelenaAungst** [comment]
ci:test:relevant

**Comment by SelenaAungst** [comment]
ci:test:platform-experience

File: modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/StyleBookManagementToolbarDisplayContext.java [hunk 1]
```diff
@@ -126,6 +144,49 @@ public Map<String, Object> getAdditionalProps() {
...
+					}
+					else {
+						CET cet = _cetManager.getCET(companyId, themeId);
+
+						if (cet != null) {
+							name = LanguageUtil.format(
+								httpServletRequest,
+								"x-theme-css-client-extension", cet.getName());
+						}
+					}
```
[the remark this answers is not in the document]
Comment (thiago-buarqque) [reply modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/StyleBookManagementToolbarDisplayContext.java:178]: @algobob  I'd say it's fine to create a method for this. But it should be called `-_getFrontendTokenDefinitionProviders`

[the remark this answers is not in the document]
Comment (SelenaAungst) [reply modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/StyleBookManagementToolbarDisplayContext.java]: Thanks @algobob @thiago-buarqque, I just pushed this change.

**Comment by SelenaAungst** [comment]
@thiago-buarqque Do I need to address the platform-experience failures here, or are test changes still not something we need to worry about now?

**Comment by thiago-buarqque** [comment]
@SelenaAungst yes, the tests are unique for your PR, meaning you have to fix them here. Regarding adding new tests, you can do it in the next PR or something

**Comment by thiago-buarqque** [comment]
ci:report:89865071

**Comment by thiago-buarqque** [comment]
ci:report:89863929

**Comment by thiago-buarqque** [comment]
ci:test:platform-experience

**Comment by thiago-buarqque** [comment]
ci:test:relevant

File: modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/StyleBookManagementToolbarDisplayContext.java [hunk 2]
```diff
@@ -166,6 +187,47 @@ public CreationMenu getCreationMenu() {
 		).build();
 	}
 
+	public List<Map<String, Object>> getFrontendTokenDefinitionProviders() {
```
Comment (algobob) [inline modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/StyleBookManagementToolbarDisplayContext.java:190]: Is this method used on other places? If not, we can make it private.

Comment (SelenaAungst) [reply modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/StyleBookManagementToolbarDisplayContext.java]: It was only used in this file, so I made the change.

**Comment by thiago-buarqque** [comment]
Resent to https://github.com/liferay-platform-experience/liferay-portal/pull/867. Please @algobob give it a final review and, when everything is fine, ask for Evan's review.

@SelenaAungst Please check my comments in each commit message.

**Pull request closed without merging**